tmt/steps/report/junit.py

```diff
@@ -193,13 +195,18 @@ def make_junit_xml(
     # Prepare the template environment
     environment = default_template_environment()
 
-    template_path = template_path or tmt.utils.resource_files(
-        DEFAULT_TEMPLATE_DIR / Path(f'{flavor}.xml.j2'))
+    template_path = template_path or DEFAULT_TEMPLATE_DIR / f'{flavor}.xml.j2'
 
     # Use a FileSystemLoader for a non-custom flavor
     if flavor != CUSTOM_FLAVOR_NAME:
+        # TODO: Check if PackageLoader would work instead
+        # Note: the issue here is that jinja passes the paths through `os.fspath` which breaks the
+        # MultiplexedPath. This can be resolved by using `as_files` to create a context and copy
+        # all files to a real data folder, or jinja could learn to support generic Traversable
+        if isinstance(DEFAULT_TEMPLATE_DIR, MultiplexedPath):
+            phase.warn("Jinja template extension for report/junit/templates is not supported yet.")
         environment.loader = FileSystemLoader(
-            searchpath=tmt.utils.resource_files(DEFAULT_TEMPLATE_DIR))
+            searchpath=str(DEFAULT_TEMPLATE_DIR))
```

Comment:
Welp, this is a nuanced place where this can break. Basically jinja does not support `MultiplexedPath` (opened an issue: https://github.com/pallets/jinja/issues/2035). I've looked a bit at the implementation and it seems it could support quite easily by just using `pathlib`/`Traversable` interface.

Alternatively this can be solved on the `tmt` side by using the `importlib.resources.as_file` context which would copy all files under the package path to a native `Path` structure. It is really not ideal and would need more refactoring of how to work with `tmt.utils.resource_files`.